### DISCUSSION TITLE
sample xonsh script to do repeatitive commands

### DISCUSSION THREAD

**Discussion by pannet1** [xonsh/xonsh#5030] [General]
i have an alias `venv` shortcut to activate my virtual env
i have 3 commands that uses the z plugin
so from the home directory i want to run the following to navigate to the directory containing my current dev files

venv
z Public
z virt
z mypyproj

how to this with xonsh script

**Comment by anki-code** [comment]
I dont understand the case. Please describe step by step

**Reply by pannet1** [reply]
hi @anki-code 
from xonsh script

how to run alias 
how to cd to a directory
something like ...

```
#!/bin/env xonsh
myalias
cd /home/sweethome
```

**Reply by anki-code** [reply]
Just use `aliases` like in interactive mode:
```xsh
echo @("""#!/bin/env xonsh
aliases['e'] = 'echo'
e hello
""") > /tmp/hello.xsh

xonsh /tmp/hello.xsh
# hello
```

**Reply by pannet1** [reply]
it is always great pleasure to interact with you sir.

how to change to a directory. should that also be an alias in order to work.

**Reply by anki-code** [reply]
If you want to change the current path from the script you need to use `source` command:
```python
cd /etc
echo 'cd /tmp' > /tmp/pwd.xsh

source /tmp/pwd.xsh

pwd
# /tmp
```

**Reply by anki-code** [reply]
Also if you're using virtual environments it's good to read: [Install xonsh with package and environment management system](https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md#install-xonsh-with-package-and-environment-management-system).

**Reply by pannet1** [reply]
got it. you always surpass the expectations. while playing with the cheat sheet  https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md#how-to-understand-the-xonsh-location 

got a surprise
![image](https://user-images.githubusercontent.com/12350276/212906445-1079834e-9742-401c-a87a-5e035b52ae3e.png)

may be its because of the xontrib xlsd

**Reply by anki-code** [reply]
> may be its because of the xontrib xlsd

yeah, try to run the same command from `xonsh --no-rc`

**Reply by pannet1** [reply]
the event loop error is fine. i have fixed it with a line in the rc.
gruvbox error i think is from kitty. will have to fix it.


![image](https://user-images.githubusercontent.com/12350276/212911105-6e92fbc7-9588-4db4-a00c-3efc5bca2370.png)

**Reply by pannet1** [reply]
cheatsheet works as expected now
![image](https://user-images.githubusercontent.com/12350276/212912640-24eeb63a-3f11-4bd8-83c7-6e82f18edbf0.png)

thank you so much

**Reply by anki-code** [reply]
I've updated the section [How to understand the xonsh location](https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md#how-to-understand-the-xonsh-location) to have more useful and clean examples.
Please take a look.

**Reply by pannet1** [reply]
your work is always xonshitive. thank you @anki-code
